code/rounds/BaseRound.cs

```diff
@@ -0,0 +1,92 @@
+using Sandbox;
+using System;
+using System.Collections.Generic;
+using TTTReborn.Player;
+
+namespace TTTReborn.Rounds
+{
+    public abstract partial class BaseRound : NetworkComponent
+    {
+        public virtual int RoundDuration => 0;
+        public virtual string RoundName => "";
+        public virtual bool CanPlayerSuicide => true;
+
+        public List<TTTPlayer> Players = new();
+
+        public float RoundEndTime { get; set; }
+
+        public float TimeLeft
+        {
+            get
+            {
+                return RoundEndTime - Sandbox.Time.Now;
+            }
+        }
+        
+        [Net] public string TimeLeftFormatted { get; set; }
+
+        public void Start()
+        {
+            if ( Host.IsServer && RoundDuration > 0 )
+            {
+                RoundEndTime = Sandbox.Time.Now + RoundDuration;
+            }
+			
+            OnStart();
+        }
+
+        public void Finish()
+        {
+            if ( Host.IsServer )
+            {
+                RoundEndTime = 0f;
+                Players.Clear();
+            }
+
+            OnFinish();
+        }
+
+        public void AddPlayer( TTTPlayer player )
+        {
+            Host.AssertServer();
+
+            if ( !Players.Contains(player) )
+            {
+                Players.Add( player );
+            }
+        }
+
+        public virtual void OnPlayerSpawn( TTTPlayer player ) { }
+
+        public virtual void OnPlayerKilled( TTTPlayer player ) { }
+
+        public virtual void OnPlayerLeave( TTTPlayer player )
```

Comment:
please don't add spaces around brackets here. I see you are inconsistent with this in your own file, probably something defined by your editor. So just a heads up

Reply:
Yup, please feel free to call out any weird coding syntax. I'm currently used to automatically formatting, so I tend to currently leave weird syntax all over the place.